core/src/main/java/hudson/search/Search.java

```diff
@@ -207,12 +207,25 @@ public static class Result {
 
     @ExportedBean(defaultVisibility = 999)
     public static class Item {
+
         @Exported
         @SuppressFBWarnings(value = "URF_UNREAD_PUBLIC_OR_PROTECTED_FIELD", justification = "read by Stapler")
         public String name;
 
+        private final String url;
+
         public Item(String name) {
+            this(name, null);
```

Comment:
Does this result in correct behavior when used? I cannot tell.

This also ends up bloating the responses of `AutocompletionCandidates`, something that looks unintentional.

Reply:
Testing the autocomplete control with the Design Library and it works fine. It does bloat the response (with a `url`) field and there will be more fields added in the future (e.g. `icon`, `group`, etc). Is this okay for now and address in a follow up MR, alternatively which method would you think best to change?

Reply:
> alternatively which method would you think best to change?

Probably subtype / separate types. Unclear why `AutocompletionCandidates` returns a `Search.Item`, seems more like out of convenience than anything else.

> Is this okay for now and address in a follow up MR

Works for me.